### PR TITLE
Fixes #107 - Add weekly report module in basic form

### LINKED ISSUE #107: Add preliminary weekly report framework for total new issues reported
We'll expand on this graph-by-graph and feature-by-feature as we go, but for now let's just get **something** up there so @miketaylr can check it off the OKRs... 😜 

First step will be to simply display the total new issues logged by week (from the GH web-bugs repo).

### PR DESCRIPTION
**Pull request by laghee**
There's obviously a lot here to be improved, but there are numbers! In a graph! With peaks and everything!

Some things to address:

- restore the possibility to filter the time period (I took it out for simplicity's sake when I thought my dates were going to be in a totally different format)
- not sure if "min" and "max" are as meaningful here
- probably lots of design and React nits 😁 

dashboard-landing
weekly-reports-tab

### CODE REVIEW

**Comment by laghee** [comment]
Ahem. So, of course, what I meant to say was, "Let me fix these tests before I go about blithely requesting reviews!"

**Comment by magsout** [comment]
@laghee awesome. so happy to review your PR

**Comment by miketaylr** [comment]
This is so cool!

**Comment by magsout** [comment]
@laghee no error at all, you just need to update the snasphots because you added some new components on existing file like the menu for example. So tests failed because there are difference between files testing and snapshots.
Just run: `npm run test:jest -- -u` after that all snapshot will be update, and all tests will be passed

File: src/containers/WeeklyReports/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+import React from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import PropTypes from "prop-types";
+
+import { getWeeklyReports } from "../../actions";
+import {
+  ObjectNested,
+  getFiltersFromUrl,
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: it's not necessary to importe this functions: `pushFiltersToUrl,` `getFiltersFormUrl`

File: src/containers/WeeklyReports/index.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import { CHART_LINE } from "../../constants/Charts";
+
+class WeeklyReports extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    /* merge filters from url and local filters */
+    const filters = {
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: you could remove this part

File: src/containers/WeeklyReports/index.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      ...this.state,
+      ...toQueryObject(getFiltersFromUrl()),
+    };
+    /* sent request */
+    this.getWeeklyReports(filters);
+    /* update view with filters */
+    this.setState({
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: you could remove this part

File: src/containers/WeeklyReports/index.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+  /* fetch data */
+  getWeeklyReports(filters = {}) {
+    /* push filters to url */
+    pushFiltersToUrl(toQueryString(filters));
+    const args = {
+      actionParameters: {
+        chartList: [CHART_LINE],
+      },
+      requestParameters: {
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: you could remove this part

File: src/containers/WeeklyReports/index.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        ...filters,
+      },
+    };
+    this.props.getWeeklyReports(args);
+  }
+
+  handleChange(e) {
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: you could remove this function

File: src/containers/WeeklyReports/index.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  handleSubmit(e) {
```
Comment (magsout) [inline src/containers/WeeklyReports/index.js]: you could remove this function

**Comment by magsout** [comment]
@laghee I think it should be a good idea to document the test side. Like something `don't forget updating snapshots if you add/edit things`

**Comment by magsout** [comment]
it's green @laghee awesome 👍

**Comment by laghee** [comment]
@magsout Yes, absolutely. That part definitely. Also, I think it would be great to have kind of a mini-tutorial or step-by-step doc that walks people through "how to add a new dashboard" in general from start to finish. (I'm trying to keep good notes for that.)

I started out by removing the date filter because I thought that would simplify things, but in the end I think it makes more sense to add that back in. Hoping to have time to do that today/tomorrow!

**Comment by laghee** [comment]
@magsout Yes, absolutely. That part definitely. Also, I think it would be great to have kind of a mini-tutorial or step-by-step doc that walks people through "how to add a new dashboard" in general from start to finish. (I'm trying to keep good notes for that.)

I started out by removing the date filter because I thought that would simplify things, but in the end I think it makes more sense to add that back in. Hoping to have time to do that today/tomorrow!

File: src/containers/WeeklyReports/index.js (hunk 1 above)
Comment (laghee) [reply src/containers/WeeklyReports/index.js]: 👍

**Comment by laghee** [comment]
@magsout OK, I think I got everything out. Let me know if I missed anything.

**Comment by magsout** [comment]
awesome

**Pull request merged**